pkg/controllers/grafana-operator/controller.go

```diff
@@ -68,13 +69,19 @@ type reconciler struct {
 	grafanaClientset rest.Interface
 }
 
+type reconcileResult struct {
```

Comment:
Should this struct and all of the related constructors be in its own package? I don't think this is specific to one controller

Reply:
I agree and I felt the same but  thought we can refactor this when/if another controller takes the same approach. But I am happy to do that in this patch if you strongly feel so.

Reply:
Ok, maybe we can just move it to a dedicated file for now